web/test/helpers-formation.tsx

```diff
@@ -76,11 +76,18 @@ export const preparePage = (
       profileData.legalStructureId as FormationLegalType
     ),
   });
-
-  const municipalitiesValue = municipalities ?? [];
-
+  const internalMunicipalities = [
+    profileData?.municipality ?? generateMunicipality({ displayName: "GenericTown" }),
+    ...(municipalities ?? []),
+  ];
+  initialUserData.formationData.formationFormData.addressMunicipality &&
+    internalMunicipalities.push(initialUserData.formationData.formationFormData.addressMunicipality);
+  initialUserData.formationData.formationFormData.agentOfficeAddressMunicipality &&
+    internalMunicipalities.push(
+      initialUserData.formationData.formationFormData.agentOfficeAddressMunicipality
+    );
```

Comment:
what is going on with this logic?

Reply:
depending on the generators, there might be 3 different generatedMunicipality objects in the data, and if they're not all passed into the MunicipalitiesContext, then the tests will throw errors as the fields are being populated with invalid data

Reply:
where do the 3 different ones come from? When i refactored the municipality context, i was explicitly passing the municipality into the array, I noticed you refactored that out?